src/core/components/dag/put.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const multicodec = require('multicodec')
+
+module.exports = ({ ipld, pin, gcLock, preload }) => {
+  return async function put (dagNode, options) {
+    options = options || {}
+
+    if (options.cid && (options.format || options.hashAlg)) {
```

Comment:
It seems like we do this hash algorithm and format selection in a couple of places in the code, I wonder if we can unify it into one central place?